src/index.js

```diff
@@ -1,390 +1,586 @@
 
-export function app(state, actions, view, container) {
-  var map = [].map
-  var rootElement = (container && container.children[0]) || null
-  var oldNode = rootElement && recycleElement(rootElement)
-  var lifecycle = []
-  var skipRender
-  var isRecycling = true
-  var globalState = clone(state)
-  var wiredActions = wireStateToActions([], globalState, clone(actions))
-
-  scheduleRender()
-
-  return wiredActions
-
-  function recycleElement(element) {
-    return {
-      nodeName: element.nodeName.toLowerCase(),
-      attributes: {},
-      children: map.call(element.childNodes, function(element) {
-        return element.nodeType === 3 // Node.TEXT_NODE
-          ? element.nodeValue
-          : recycleElement(element)
-      })
+      if (
+        name in element &&
+        name !== "list" &&
+        name !== "draggable" &&
+        name !== "spellcheck" &&
+        name !== "translate" &&
+        !isSvg
+      ) {
+        element[name] = nextValue == null ? "" : nextValue
+        if (nullOrFalse) {
+          element.removeAttribute(name)
+        }
+      } else {
+        var ns = isSvg && name !== (name = name.replace(/^xlink:?/, ""))
+        if (ns) {
+          if (nullOrFalse) {
+            element.removeAttributeNS(XLINK_NS, name)
+          } else {
+            element.setAttributeNS(XLINK_NS, name, nextValue)
+          }
+        } else {
+          if (nullOrFalse) {
+            element.removeAttribute(name)
+          } else {
+            element.setAttribute(name, nextValue)
+          }
+        }
+      }
     }
   }
+}
 
-  function resolveNode(node) {
-    return typeof node === "function"
-      ? resolveNode(node(globalState, wiredActions))
-      : node != null
-        ? node
-        : ""
+var createElement = function(node, lifecycle, eventProxy, isSvg) {
+  var element =
+    node.type === TEXT_NODE
+      ? document.createTextNode(node.name)
+      : (isSvg = isSvg || node.name === "svg")
+        ? document.createElementNS(SVG_NS, node.name)
+        : document.createElement(node.name)
+
+  var props = node.props
+  if (props.onCreate) {
+    lifecycle.push(function() {
+      props.onCreate(element)
+    })
   }
 
-  function render() {
-    skipRender = !skipRender
+  for (var i = 0, length = node.children.length; i < length; i++) {
+    element.appendChild(
+      createElement(node.children[i], lifecycle, eventProxy, isSvg)
+    )
+  }
 
-    var node = resolveNode(view)
+  for (var name in props) {
+    updateProperty(element, name, null, props[name], eventProxy, isSvg)
+  }
+
+  return (node.element = element)
+}
 
-    if (container && !skipRender) {
-      rootElement = patch(container, rootElement, oldNode, (oldNode = node))
+var updateElement = function(
+  element,
+  lastProps,
+  nextProps,
+  lifecycle,
+  eventProxy,
+  isSvg,
+  isRecycled
+) {
+  for (var name in merge(lastProps, nextProps)) {
+    if (
+      (name === "value" || name === "checked"
```

Comment:
Just a matter of taste, i like single quotes for strings in JS and double quotes inside JSX/HTML.

Reply:
It's not bad suggestion, in fact I prefer single quotes for strings too.

Reply:
But in that case, It's  Hyperapp @jorgebucaran options :sweat_smile: